awriter/writer_test.go

```diff
@@ -111,6 +111,23 @@ func TestWriteArtifactWithUpdates(t *testing.T) {
 	assert.NoError(t, err)
 
 	assert.NoError(t, checkTarElements(buf, 3))
+
+	// Update with invalid data file name.
+	upd, err = MakeFakeInvalidUpdate("my test update")
+	assert.NoError(t, err)
+
+	u = handlers.NewRootfsV1(upd)
+	updates = &Updates{Updates: []handlers.Composer{u}}
+
+	err = w.WriteArtifact(&WriteArtifactArgs{
+		Format:  "mender",
+		Version: 1,
+		Devices: []string{"asd"},
+		Name:    "name",
+		Updates: updates,
+	})
+
+	assert.Error(t, err)
```

Comment:
In `reader_test.go` majority of tests use `MakeRootfsImageArtifact` method to create an artifact. As a result method `MakeAnyImageArtifact` is invoked where `WriteArtifact` from `awriter` is called and in consequence `writeOneDataFile` is called where invalid characters are checked. So the test will fail in the moment of artifact building, not in time of reading it in `ReadArtifact` method.
The question is whether we want each test be duplicated just to ignore an error reported by `MakeRootfsImageArtifact` and focus on error detection from `ReadArtifact`?

Reply:
There are already some tests that deal with broken artifacts. See if you can use something from [here](https://github.com/mendersoftware/mender-artifact/blob/mender-artifact-v3/areader/reader_test.go#L680-L709). It uses the `manipulateArtifact` function field to make arbitrary changes to the artifact and then manually reassemble it, even if it's corrupted. In your case you'd want to add a new test case where you change the file name and the line containing the filename inside the `manifest` file.

I think reading is quite important to test because this is a potential security issue, and even if we don't write such an artifact, someone could manually craft one that has the illegal characters.

Reply:
Sure, will do.